File: examples/typescript-sdk/context/prompt-enhancer-server/response-parser.ts
// Regex for extracting enhanced prompt from AI response
const ENHANCED_PROMPT_REGEX =
  /<enhanced-prompt>([\s\S]*?)<\/enhanced-prompt>/;

/**
 * Parse the enhanced prompt from the AI response
 */
export function parseEnhancedPrompt(response: string): string | null {
  // Extract content between <augment-enhanced-prompt> tags
  const match = response.match(ENHANCED_PROMPT_REGEX);

  if (match?.[1]) {
    return match[1].trim();
  }

  return null;
}

File: examples/python-sdk/context/prompt_enhancer_server/response_parser.py
"""Response parser for extracting enhanced prompts from AI responses"""

import re

# Regex for extracting enhanced prompt from AI response
ENHANCED_PROMPT_REGEX = re.compile(r"<enhanced-prompt>(.*?)</enhanced-prompt>", re.DOTALL)


def parse_enhanced_prompt(response: str) -> str | None:
    """
    Parse the enhanced prompt from the AI response

    Args:
        response: The AI response containing the enhanced prompt

    Returns:
        The enhanced prompt text, or None if not found
    """
    # Extract content between <enhanced-prompt> tags
    match = ENHANCED_PROMPT_REGEX.search(response)

    if match and match.group(1):
        return match.group(1).strip()

    return None

